lib/options/sender_option_parser.rb

```diff
@@ -255,30 +277,49 @@ def initialize(args)
 
       # Parse basic, common and specific options for sender client
       parse(args)
-    end
+      # Apply options which need to be applied last
+      @doLast.each { |proc| proc.call }
+    end # initialize
 
     private
     def parse_kv(key_value)
       if key_value.include? "="
         key, value = key_value.split("=")
         value = "" if value.nil?
+        return key, manual_cast(@options.content_type, value)
       elsif key_value.include? "~"
         key, value = key_value.split("~")
-        value = auto_cast(value)
+        return key, auto_cast(value)
       else
         raise OptionParser::InvalidArgument, "kv pair '#{key_value}' is of invalid format, = or ~ required"
       end
-      return key, value
     end
 
+    private
     def auto_cast(value)
-      if value.include? "."
-        value = value.to_f
-      else
-        value = value.to_i
+      if StringUtils.str_is_int?(value)
+        return value.to_i
+      elsif StringUtils.str_is_float?(value)
+        return value.to_f
       end
       value
-    end # initialize
+    end
+
+    private
+    def manual_cast(type, value)
+      case type
+      when "int"
+        return value.to_i
```

Comment:
return can be omitted.

to_i is not throwing exception when whole value does not contain integer, so "123jarda" will be 123, hmm.

Reply:
`Integer(value)`